extract case update code to separate class

snopoke opened the pull request:
@benrudolph 

Just moving code around. Extracting the code that updates a case into its own class.

This is a PR into #8644 

cc @millerdev

agrzywinski committed:
Migrate email report settings for users

agrzywinski committed:
test fix

agrzywinski committed:
Merge branch 'master' into 162237

Conflicts:
	custom/ewsghana/api.py

Commit:
Merge remote-tracking branch 'origin/162237' into email-report

Conflicts:
	custom/ewsghana/api.py
	custom/ewsghana/tests/test_apis.py

Commit:
Added queues for logistics tasks

Commit:
bower_components cache

snopoke committed:
extract case update code to separate class

dimagimon commented on corehq/ex-submodules/casexml/apps/case/update_strategy.py:
```diff
@@ -0,0 +1,425 @@
+        case = CommCareCase()
+        case._id = case_update.id
+        case.modified_on = parsing.string_to_utc_datetime(case_update.modified_on_str) \
+                            if case_update.modified_on_str else datetime.utcnow()
```
E127 continuation line over-indented for visual indent

dimagimon commented on corehq/ex-submodules/casexml/apps/case/update_strategy.py:
```diff
@@ -0,0 +1,425 @@
+        """
+        Clear known case properties, and all dynamic properties
+        """
+        dynamic_properties = set([k for action in self.case.actions for k in action.updated_unknown_properties.keys()])
```
E501 line too long (119 > 115 characters)

millerdev commented:
:+1: nice cleanup

Commit:
Exclude email settings from steady sync

NoahCarnahan committed:
Add versions to media_suite.xml resource elements

Commit:
Merge branch 'master' into bower-travis

kaapstorm committed:
Drop absolute position and put form data inside container

Commit:
Add SMSGH Gateway

Commit:
Add tests

Commit:
[ci skip] Remove unnecessary line

Commit:
add description field to ReportAppConfig

Commit:
default to the report's description

Commit:
allow user to customize the description

Commit:
use localized, customized description in the app

Commit:
use localized, customized title in the app

Commit:
Merge branch 'master' into smsgh

Commit:
memoize indicators property

benrudolph committed:
Merge pull request #8653 from dimagi/smsgh

Add SMSGH Backend

benrudolph committed:
Merge pull request #8655 from dimagi/memoize-indicators

memoize indicators property

Commit:
Properly escape js string

Commit:
escapejs on mobile workers page

orangejenny committed:
Merge pull request #8654 from dimagi/editable-description

Edit description of mobile report

benrudolph committed:
Revert "FileDownload for ExportSchema"

biyeun committed:
Merge pull request #8652 from dimagi/fb/183533/css

Drop absolute position and put form data inside container

benrudolph committed:
Always check celery on server.txt

@czue, realized that our main pingdom hits server.txt and our rabbitMQ one hits server.txt?celery=true. thinking this would be the way to consolidate unless we feel that's there a reason to not do this. maybe celery goes down more often so it caused too much noise for the main pingdom?

benrudolph commented:
makes sense nice change

benrudolph commented:
(unmergable)

czue committed:
Merge pull request #8656 from dimagi/escape-js-trans

Escape js trans

czue committed:
Merge pull request #8657 from dimagi/revert-8584-enable-dropbox-export

Revert "FileDownload for ExportSchema"

czue committed:
Merge pull request #8659 from dimagi/benrudolph-patch-1

Always check celery on server.txt

czue commented:
:walking: :walking:

czue commented:
(this is great)

czue committed:
Merge pull request #8630 from dimagi/bower-travis

bower_components cache

dannyroberts committed:
Merge pull request #8649 from dimagi/cli-fix

Tweak "view source" page for latest version of of apps

NoahCarnahan committed:
Revert "Tweak "view source" page for latest version of of apps"

NoahCarnahan committed:
Merge pull request #8660 from dimagi/revert-8649-cli-fix

Revert "Tweak "view source" page for latest version of of apps"

dannyroberts committed:
fix deleting docs in continuous doctype migration

deleted docs in the changes feed come back with a theoretical rev that
isn't useful for deleting in another db. Fetch the rev from the target
db instead.

snopoke committed:
remove unused (duplicated) method

dannyroberts committed:
doctypemigrations: add --erase_continuous_progress

for when something has gone wrong with the continuous migration
and you need to reset the progress and start from the original seq

NoahCarnahan committed:
update submodule

czue committed:
Merge pull request #8663 from dimagi/fix-continuous-delete

Fix continuous delete

dannyroberts committed:
update to commcare-translations master

czue committed:
Merge pull request #8664 from dimagi/update-formtranslate

Update formtranslate submodule

czue committed:
Merge pull request #8592 from lwyszomi/email-report

EWS: Email reports

czue committed:
Merge pull request #8620 from lwyszomi/logistics-queues

Added queues for logistics tasks

czue committed:
Merge pull request #8661 from dimagi/cull

remove unused (duplicated) method

snopoke committed:
Merge pull request #8644 from dimagi/common-test-continue

case cleanup

snopoke commented:
closed in favour of https://github.com/dimagi/commcare-hq/pull/8667
